engine/consensus/matching/engine.go

```diff
@@ -1225,9 +1303,35 @@ func (e *Engine) requestPendingApprovals() (int, error) {
 		}
 	}
 
-	if requestCount > 0 {
-		log.Debug().Msgf("requested %d approvals", requestCount)
+	return requestCount, nil
+}
+
+// OnFinalizedBlock implements the callback from the protocol state to notify a block
+// is finalized, it guarantees every finalized block will be called at least once.
+func (e *Engine) OnFinalizedBlock(block *model.Block) {
+	// we index the execution receipts by the executed block ID only for all finalized blocks
+	// that guarantees if we could retrieve the receipt by the index, then the receipts
+	// must be for a finalized blocks.
+	err := e.indexReceipts(block.BlockID)
+	if err != nil {
+		e.log.Fatal().Err(err).Hex("block_id", block.BlockID[:]).
+			Msg("could not index receipts for block")
+	}
+}
```

Comment:
we need to subscribe to the block finalization event to index all receipts by the executed block in order for this to be used in checking sealing